src/Umbraco.Web/Routing/ContentFinderByIdPath.cs

```diff
@@ -54,14 +54,40 @@ public bool TryFindContent(PublishedRequest frequest)
 
                     if (node != null)
                     {
+                        string culture = null;
```

Comment:
Looking at the code below, the culture will never be null and this would be a lot easier to read, am I missing something?

```csharp
var culture = frequest.Culture.Name;
//if we have a node, check if we have a culture in the query string
if (frequest.UmbracoContext.HttpContext.Request.QueryString.ContainsKey("culture"))
{
    culture = frequest.UmbracoContext.HttpContext.Request.QueryString["culture"];

    if (culture != "invariant")
    {
        //we're assuming it will match a culture, if an invalid one is passed in, an exception will throw (there is no TryGetCultureInfo method), i think this is ok though
        frequest.Culture = CultureInfo.GetCultureInfo(frequest.UmbracoContext.HttpContext.Request.QueryString["culture"]);
    }
}
```

Reply:
Updated